src/System.Net.HttpListener/src/System/Net/WebSockets/HttpWebSocket.cs

```diff
@@ -167,7 +167,21 @@ private static void ValidateWebSocketHeaders(HttpListenerContext context)
                     SupportedVersion));
             }
 
-            if (string.IsNullOrWhiteSpace(context.Request.Headers[HttpKnownHeaderNames.SecWebSocketKey]))
+            string secWebSocketKey = context.Request.Headers[HttpKnownHeaderNames.SecWebSocketKey];
+            bool isSecWebSocketKeyInvalid = string.IsNullOrWhiteSpace(secWebSocketKey);
+            if (!isSecWebSocketKeyInvalid)
+            {
+                try
+                {
+                    // key must be 16 bytes then base64-encoded
+                    isSecWebSocketKeyInvalid = Convert.FromBase64String(secWebSocketKey.Trim()).Length != 16;
```

Comment:
Does this need to be trimmed? I thought HeadersValueCollection (or whatever the type context.Request.Headers) automatically trims content